terraform-continuous-deployment/terraform-continuous-deployment.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: v1
+description: Automatically run a Terraform plan when a GitHub PR is merged
+
+triggers:
+- name: github-pr-merge
+  source:
+    type: webhook
+    image: relaysh/github-trigger-pull-request-merged:latest
+  binding:
+    parameters:
+      repository: !Data repository
+      branch: !Data branch
+  when:
+  - !Fn.equals [!Data branch, "master"]
+
+parameters:
+  repository:
+    description: The name of the repository that had a PR merged
+  branch:
+    description: The branch that the PR was pulled into
+    default: master
+  terraformWorkspace:
+    description: The Terraform workspace to use
+    default: default
+  terraformPlanDir:
+    description: The directory of the Terraform plan
+  terraformCommand:
+    description: The Terraform command to run
+    default: apply
+  terraformPlanRepository:
+    description: The repository that holds the terraform plan
+  terraformPlanRepositoryBranch:
+    description: The repository branch to use when cloning the repository
+    default: master
+
+steps:
+- name: terraform-run
```

Comment:
I think we were going to do plan -> approval -> apply as the steps here, right? Or is that not possible in our current system?

Reply:
Having an approval seems to defeat the purpose of CD on the trigger. At that point, why wouldn't someone just click "Run" in the UI? I do like the idea of running `plan` first to dump what's going to happen. What do you think?

Reply:
Yeah, I feel ya. What if we put the approval in, but commented out, as an alternative way of running it? I think if our approvals supported email notifications it would make more sense to have it as part of an automated process.